packages/backend/src/managers/applicationManager.ts

```diff
@@ -132,15 +132,21 @@ export class ApplicationManager {
       throw new Error('Cannot load configuration file.');
     }
 
-    // Mark as success.
-    loadingConfiguration.state = 'success';
-    taskUtil.setTask(loadingConfiguration);
-
     // Filter the containers based on architecture
     const filteredContainers = aiConfig.application.containers.filter(
       container => container.arch === undefined || container.arch === arch(),
     );
 
+    if (filteredContainers.length > 0) {
+      // Mark as success.
+      loadingConfiguration.state = 'success';
+      taskUtil.setTask(loadingConfiguration);
+    } else {
+      // Mark as failure.
+      loadingConfiguration.state = 'error';
+      taskUtil.setTask(loadingConfiguration);
+    }
+
```

Comment:
line 143 and 147looks the same so probably should be after the if/else

Reply:
Since I am throwing an error I can't

Reply:
before there was not the throw line 🤣 

but you could do to avoid duplicating lines

```ts
try {
  if (filteredContainers.length > 0) {
    // Mark as success.
    loadingConfiguration.state = 'success';
  } else {
    // Mark as failure.
    loadingConfiguration.state = 'error';
    throw new Error('No containers available.');
  }
} finally {
  taskUtil.setTask(loadingConfiguration);
}
```

(you can of course discard my comment :-)

Reply:
Yeah sorry I should have said I updated the code ahah.

I don't mind having one line duplicates instead of adding a try finally